packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -586,153 +637,525 @@ async function handler(req: NextApiRequest & { userId?: number | undefined }) {
+        url: videoCallReference.meetingUrl,
+      };
+    }
+  };
+
+  /* Check if the original booking has no more attendees, if so delete the booking 
+  and any calendar or video integrations */
+  const lastAttendeeDeleteBooking = async (
+    originalRescheduledBooking: Awaited<ReturnType<typeof getOriginalRescheduledBooking>>,
+    originalBookingEvt?: CalendarEvent
+  ) => {
+    if (
+      originalRescheduledBooking?.attendees.filter((attendee) => {
+        return attendee.email !== reqBody.email;
+      }).length === 0
+    ) {
+      const integrationsToDelete = [];
+
+      for (const reference of originalRescheduledBooking.references) {
+        if (reference.credentialId) {
+          const credential = await prisma.credential.findUnique({
+            where: {
+              id: reference.credentialId,
+            },
+          });
+
+          if (credential) {
+            if (reference.type.includes("_video")) {
+              integrationsToDelete.push(deleteMeeting(credential, reference.uid));
+            }
+            if (reference.type.includes("_calendar") && originalBookingEvt) {
+              const calendar = getCalendar(credential);
+              if (calendar) {
+                integrationsToDelete.push(
+                  calendar?.deleteEvent(reference.uid, originalBookingEvt, reference.externalCalendarId)
+                );
+              }
+            }
+          }
+        }
+      }
+
+      await Promise.all(integrationsToDelete).then(async () => {
+        await prisma.booking.delete({
+          where: {
+            id: originalRescheduledBooking.id,
+          },
+        });
+      });
+    }
+  };
+
+  const handleSeats = async () => {
+    // For seats, if the booking already exists then we want to add the new attendee to the existing booking
     if (!eventType.seatsPerTimeSlot) {
       throw new HttpError({ statusCode: 404, message: "Event type does not have seats" });
     }
 
     const booking = await prisma.booking.findUnique({
       where: {
-        uid: reqBody.bookingUid,
+        uid: rescheduleUid || reqBody.bookingUid,
       },
       select: {
         uid: true,
         id: true,
-        attendees: true,
+        attendees: { include: { bookingSeatReference: true } },
         userId: true,
         references: true,
         startTime: true,
         user: true,
+        status: true,
       },
     });
     if (!booking) {
       throw new HttpError({ statusCode: 404, message: "Booking not found" });
     }
 
-    // Need to add translation for attendees to pass type checks. Since these values are never written to the db we can just use the new attendee language
-    const bookingAttendees = booking.attendees.map((attendee) => {
-      return { ...attendee, language: { translate: tAttendees, locale: language ?? "en" } };
-    });
+    if (dayjs(booking.startTime).utc().format() === evt.startTime) {
+      return booking;
+    }
```

Comment:
If we are rescheduling and the same timeslot is chosen then just return the booking